src/qt/diagnose.h

```diff
@@ -0,0 +1,627 @@
+{
+public:
+    static bool m_researcher_mode;
+    static const ResearcherModel* m_researcher_model;
+
+    enum diagnoseResults { PASS,
+                           WARNING,
+                           FAIL,
+                           NONE };
+    enum TestNames {
+        VerifyWalletIsSynced,
+        CheckClientVersion,
+        CheckConnectionCount,
+        CheckOutboundConnectionCount,
+        VerifyBoincPath,
+        VerifyCPIDHasRAC,
+        VerifyCPIDIsActive,
+        VerifyCPIDValid,
+        VerifyClock,
+        VerifyTCPPort,
+        CheckDifficulty,
+        TestSize // add any new test before this entry
+    };
+
+    Diagnose()
+    {
+        m_results = NONE;
+        m_results_string = "";
+        m_researcher_mode = false;
+        registerTest(this);
+    }
+    virtual ~Diagnose()
+    {
+        LOCK(cs_diagnostictests);
+        removeTestFromMap(m_test_name);
+    }
+
+    /* runCheck(): calling the function will run the test*/
+    virtual void runCheck() = 0;
+    /*Get teh result of test , Fail, Warning, or PAss*/
+    virtual diagnoseResults getResults() { return m_results; }
+    /*Return a string containing a tip for the test being done*/
+    virtual std::string getResultsTip() { return m_results_tip; }
+    /*return the final result string*/
+    virtual std::string getResultsString() { return m_results_string; }
+    /*The result can contains arguments using "$1". The functions returns the strings that should replace the argument*/
+    virtual std::vector<std::string> getStringArgs() { return m_results_string_arg; }
+    /*The Tip can contains arguments using "$1". The functions returns the strings that should replace the argument*/
+    virtual std::vector<std::string> getTipArgs() { return m_results_tip_arg; }
+    /*Register a running test to the map so we can check if it is runnin or not.*/
+    static void registerTest(Diagnose* test)
+    {
+        LOCK(cs_diagnostictests);
+        m_name_to_test_map[test->m_test_name] = test;
+        assert(m_name_to_test_map.size() < Diagnose::TestSize);
+    };
+    /*Set the research mode the wallet is running */
+    static void setResearcherModel(ResearcherModel* model)
+    {
+        m_researcher_model = model;
+        m_researcher_mode = !(m_researcher_model->configuredForInvestorMode() || m_researcher_model->detectedPoolMode());
+    }
+    /*Get the research mode used during testing */
+    static bool getResearcherModel() { return m_researcher_model; }
+
+    /*Get a pointer to the test object if it exists
+     * retun nullptr if no test available
+     */
+    static Diagnose* getTest(Diagnose::TestNames test_name)
+    {
+        LOCK(cs_diagnostictests);
+        auto entry = m_name_to_test_map.find(test_name);
+        if (entry != m_name_to_test_map.end()) {
+            return entry->second;
+        } else {
+            return nullptr;
+        }
+    }
+    // remove the test from the globa map so it can not be tracked any more
```

Comment:
"globa" should be "global"

Reply:
Done

Reply:
Not done. Looks like you may have lost a commit.